src/Symfony/Component/Form/FormOrderer.php

```diff
@@ -0,0 +1,327 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Form;
+
+use Symfony\Component\Form\Exception\InvalidConfigurationException;
+
+/**
+ * Form orderer.
+ *
+ * @author GeLo <geloen.eric@gmail.com>
+ */
+class FormOrderer
+{
+    /**
+     * @var array
+     */
+    private $weights;
+
+    /**
+     * @var array
+     */
+    private $deferred;
+
+    /**
+     * @var int
+     */
+    private $firstWeight;
+
+    /**
+     * @var int
+     */
+    private $currentWeight;
+
+    /**
+     * @var int
+     */
+    private $lastWeight;
+
+    /**
+     * Orders the form.
+     *
+     * @param FormInterface $form The form.
+     *
+     * @return array The ordered form child names.
+     *
+     * @throws \Symfony\Component\Form\Exception\InvalidConfigurationException If a position is not valid.
+     */
+    public function order(FormInterface $form)
+    {
+        $this->reset();
+
+        foreach ($form as $child) {
+            $position = $child->getConfig()->getPosition();
+
+            if (empty($position)) {
+                $this->processEmptyPosition($child);
+            } elseif (is_string($position)) {
+                $this->processStringPosition($child, $position);
+            } else {
+                $this->processArrayPosition($child, $position);
+            }
+        }
+
+        asort($this->weights, SORT_NUMERIC);
+
+        return array_keys($this->weights);
+    }
+
+    /**
+     * Process the form using the current weight in order to maintain the default order.
+     *
+     * @param FormInterface $form The form.
```

Comment:
All phpdocs that does not add something not contained in the code like this one should be removed.